pytorch_lightning/utilities/enums.py

```diff
@@ -143,24 +143,25 @@ def deprecate(self) -> None:
         )
 
 
-class DeviceType(LightningEnum):
-    """Define Device type by its nature - acceleatrors.
+class DeviceType(LightningEnum, metaclass=_OnAccessEnumMeta):
+    """Define Device type by its nature - accelerators.
 
-    >>> DeviceType.CPU == DeviceType.from_str('cpu')
-    True
-    >>> # you can match the type with string
-    >>> DeviceType.GPU == 'GPU'
-    True
-    >>> # which is case invariant
-    >>> DeviceType.TPU in ('tpu', 'CPU')
-    True
+    Deprecated since v1.6.0 and will be removed in v1.8.0.
+
+    Use `_AcceleratorType` instead.
     """
 
     CPU = "CPU"
     GPU = "GPU"
     IPU = "IPU"
     TPU = "TPU"
 
+    def deprecate(self) -> None:
+        rank_zero_deprecation(
+            "`DeviceType` Enum has been deprecated in v1.6 and will be removed in v1.8."
+            " Use the string value `{self.value!r}` instead."
```

Comment:
I don't understand this. Shouldn't it be "use _AcceleratorType instead"? Or rather delete this, because there is no real alternative on the public api side.

Reply:
We tell them to use the value instead directly. Tbh, I think it makes sense